src/types.ts

```diff
@@ -290,3 +388,18 @@ export interface TurboUnauthenticatedClientInterface
 export interface TurboAuthenticatedClientInterface
   extends TurboAuthenticatedPaymentServiceInterface,
     TurboAuthenticatedUploadServiceInterface {}
+
+export type TokenCreateTxParams = {
+  target: string;
+  tokenAmount: BigNumber;
+  feeMultiplier: number;
+};
+
+export type BaseTx = { id: string; target: string; reward: string };
+
+export interface BaseToken<T extends BaseTx = BaseTx> {
```

Comment:
Nit: Base to me usually implies a class hierarchy. Consider `TokenLike<>`, `Token<>` (maybe too generic), `BlockchainToken<>`, etc.

Reply:
Actually looking at the interface, it seems more network oriented... Maybe `TokenNetwork` or `TokenTools`?

Reply:
1bf3810